File: InsertionSort.js
## Insertion sort Js

function insertionSort(items) {
  for (var i = 0; i < items.length; i++) {
    let value = items[i]
    let j = i - 1
    while ( j > -1 && items[j] > value) {
    
      items[j + 1] = items[j]
      j--;
    }
  
    items[j + 1] = value
    
  }

  return list
}

const list = [54, 26, 93, 17, 77, 31, 44, 55, 20]
console.log(insertionSort(list))   // sorted [ 17, 20, 26, 31, 44, 54, 55, 77, 93 ]

File: Insertion sort/InsertionSort.py
def InsertionSort(L):
  n = len(L)
  for j in range(1,n):
    key = L[j]
    i = j - 1

    while i >= 0 and L[i] > key:
      L[i+1] = L[i]
      i = i - 1
    L[i+1] = key
  
if __name__ == "__main__":
  L = [5, 2, 6, 1, 3, 4 ]
  print("The Unsorted Array: ",L)
  InsertionSort(L)
  print("The Sorted Array: ",L)
